net/strongswan/files/ipsec.init

```diff
@@ -244,7 +248,7 @@ config_remote() {
 		local ipdest
 
 		[ "$remote_gateway" = "%any" ] && ipdest="1.1.1.1" || ipdest="$remote_gateway"
-		local_gateway=`ip route get $ipdest | awk -F"src" '/src/{gsub(/ /,"");print $2}'`
+		local_gateway="$(ip -j route get "$ipdest" | jsonfilter -e '@[0].gateway')"
```

Comment:
What does this change fix?

I have mixed feelings about having JSON become a core dependency for a lot of packages...

Reply:
The first patch fixes a bug when local_identifier is not set.
Since 19.07, 'ip route get' has an 'uid 0' after the src address.
This was appended to local ip address as in '192.168.1.1uid0'

Without json you cannot even boot OpenWrt. It is a core requirement.
It will only break when someone implement a new ip variant, like busybox, that
both remove json support and satisfy all other strongswan requirement.  But that's a problem
for who propose that new ip command.

I did implemented a solution without json. See 4d934ed27060947e62935e5463bf51cd39b82888